luarules/gadgets/unit_objectify.lua

```diff
@@ -35,6 +35,15 @@ if gadgetHandler:IsSyncedCode() then
         end
     end
 
+    function gadget:UnitGiven(unitID, unitDefID, oldTeam, newTeam)
+        if isObject[unitDefID] and Spring.ValidUnitID(unitID) then
+            Spring.SetUnitStealth(unitID, true)
+            Spring.SetUnitSonarStealth(unitID, true)
+            Spring.SetUnitNeutral(unitID, true)
+            Spring.SetUnitBlocking(unitID, true, true, true, true, true, true, false) -- set as crushable
+        end
```

Comment:
This should be extracted to a function called from both `UnitCreated` and `UnitGiven`.

Reply:
Done